@xen-orchestra/proxy/src/app/mixins/api.js

```diff
@@ -67,7 +68,7 @@ export default class Api {
         const { method, params } = body
         warn('call error', { method, params, error })
         ctx.set('Content-Type', 'application/json')
-        ctx.body = format.error(body.id, error)
+        ctx.body = format.error(body.id, new JsonRpcError(error.message, error.code))
```

Comment:
This does not respect the security principle of not sending the error to the client unless specifically authorized.

Reply:
Please explain me why we are using `format.error` if we won't the client to be aware of the error? And what are the errors that should be authorized and the errors that shouldn't be?

Thank you :)

Reply:
`format.error()` is specially designed to replace any errors that do not inherit from `JsonRpcError` or implement `toJsonRpcError()` by a generic error.

This protects from sending sensitive info to the client but at the same time allows the dev to expose API errors :slightly_smiling_face: